hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/chartComponent/pieChartComponent.tsx

```diff
@@ -0,0 +1,138 @@
+    }) {
+        super(props);
+        this.state = {
+            windowWidth: window.innerWidth,
+            windowHeight: window.innerHeight,
+            labelLengthOverflow: false
+        };
+    }
+
+    updateWindowSize = () => {
+        this.setState({
+            ...this.state,
+            windowHeight: window.innerHeight,
+            windowWidth: window.innerWidth
+        });
+    }
+
+    parseLabels = (label: string, charlen: number) => {
+        let res = "";
+        while(label.length > 0){
+            res += label.substring(0, charlen) + "<br>";
+            label = label.substring(charlen)
+        }
+        return res;
+    }
+
+    componentDidMount(): void {
+        window.addEventListener('resize', this.updateWindowSize);
+        this.props.plotData.forEach((data) => {
+            if (data && Object.keys(data).length !== 0){
+                if (data["labels"].some((label: string) => { return label.length > 35 })){
+                    this.setState({
+                        ...this.state,
+                        labelLengthOverflow: true
+                    })
+                }
+            }
+        });
+    }
+
+    componentWillUnmount(): void {
+        window.removeEventListener('resize', this.updateWindowSize);
+    }
+
+    render() {
+        const { windowHeight, windowWidth, labelLengthOverflow } = this.state;
+        let layoutProps: Partial<Plotly.Legend> = 
+        {
+            width: windowWidth * 0.8,
+            height: windowHeight - 100,
+            font: {
+                family: 'Roboto, sans-serif',
+                size: windowWidth * 0.008
+            },
+            showlegend: true,
+            legend: {
+                font: {
+                    size: windowWidth * 0.009
+                },
+                y: - windowHeight - 300,
+                yanchor: "bottom"
+            },
+            title: {
+                text: this.props.title,
+                font: {
+                    size: 20
+                }
+            },
+            margin: {
+                l: labelLengthOverflow ? windowWidth * 0.3 : 0,
+                r: (labelLengthOverflow && windowWidth < 1200) ? windowWidth * 0.2 : 0,
+                b: 100
+            },
+            paper_bgcolor: "#FFFFFF"
+        }
+        if (windowWidth < 1200) {
+            // We are now almost at tablet/small size screen
+            layoutProps["width"] = windowWidth * 0.7
+            layoutProps["height"] = windowHeight * 0.9
+            layoutProps["legend"]["orientation"] = "h"
+            // layoutProps["width"] = windowWidth * 
+            layoutProps["legend"]["x"] = 0.4;
+            layoutProps["legend"]["xanchor"] = "center";
+            layoutProps["legend"]["y"] = (- windowHeight);
+            layoutProps["legend"]["yanchor"] = "bottom";
+            layoutProps["legend"]["font"]["size"] = windowWidth * 0.009;
+            layoutProps["margin"]["b"] = windowHeight * 0.01
+        }
```

Comment:
Nit: I think it will help readability if we initialize `layoutProps` with the properties that are unchanged regardless of window size(e.g. `showlegend: true`, `paper_bgcolor`, etc) and update the changing `layoutProps` properties in the if case statements

```
if (windowWidth < 1200) {
   // Update layout props for small screen
} else {
  // Update layout props for large screen
}
```
Also the commented code `layoutProps["width"]...` can be removed if not needed.

Reply:
I think we are already initialising layoutProps with the default values.
Could you check [this](https://github.com/apache/ozone/blob/16a08005fb742ed1dda526af631b20ff4a80b6bf/hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/chartComponent/pieChartComponent.tsx#L86) and let me know if it is what you meant?

Removed the unused comment line in the latest commit.

Reply:
> Nit: I think it will help readability if we initialize layoutProps with the properties that are unchanged regardless of window size(e.g. showlegend: true, paper_bgcolor, etc) and update the changing layoutProps properties in the if case statements

I think you can ignore this. The current code is fine as it is.

> Removed the unused comment line in the latest commit.

Thanks for the update.